### PR TITLE
Serial ops

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/drivers/pl011.c [hunk 1]
```diff
@@ -160,8 +178,8 @@ void pl011_init(struct pl011_data *pd, paddr_t pbase, uint32_t uart_clk,
 	/* Configure TX to 8 bits, 1 stop bit, no parity, fifo disabled. */
 	io_write32(base + UART_LCR_H, UART_LCRH_WLEN_8);
 
-	/* Enable interrupts for receive and receive timeout */
-	io_write32(base + UART_IMSC, UART_IMSC_RXIM | UART_IMSC_RTIM);
+	/* Enable receive interrupt */
+	io_write32(base + UART_IMSC, UART_IMSC_RXIM);
```
Comment (jforissier) [inline core/drivers/pl011.c:182]: Unrelated change?

File: core/arch/arm/plat-vexpress/main.c [hunk 2]
```diff
@@ -114,14 +114,14 @@ static void read_console(void)
...
-static enum itr_return console_itr_cb(struct itr_handler *hdl)
+static enum itr_return console_itr_cb(struct itr_handler *hdl __unused)
 {
 	if (notif_async_is_started()) {
 		/*
 		 * Asynchronous notifications are enabled, lets read from
 		 * uart in the bottom half instead.
 		 */
-		interrupt_disable(hdl->chip, hdl->it);
+		console_data.chip.ops->rx_intr_disable(&console_data.chip);
```
Comment (jforissier) [inline core/arch/arm/plat-vexpress/main.c:124]: Having two abbreviations for "interrupt" (itr and intr) is somewhat unfortunate but I suppose it's OK for now.

**Review by jforissier — COMMENTED** [review]
- For "core: serial: add rx_intr_{enable,disable}() callbacks":
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
- For "core: pl011: implement rx_intr_{enable,disable}() callbacks":
  One comment, then:
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
- For "plat-vexpress: use serial callbacks rx_intr_{en,dis}able()":
  `Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Updated the commit message for "core: pl011: implement rx_intr_{enable,disable}() callbacks".

**Comment by jenswi-linaro** [comment]
Tags applied
